liota/lib/transports/amqp.py

```diff
@@ -0,0 +1,343 @@
+                raise ValueError("Error : Client key found, but client certificate not found")
+            if client_cert_available and not client_key_available:
+                log.error("Error : Client key found, but client certificate not found")
+                raise ValueError("Error : Client certificate found, but client key not found")
+
+            """
+            Setup ssl options
+            """
+            ssl_details = ({"ca_certs": self.remote_system_identity.root_ca_cert,
+                            "certfile": self.edge_system_identity.cert_file,
+                            "keyfile": self.edge_system_identity.key_file,
+                            "cert_reqs": getattr(ssl, self.tls_details.cert_required),
+                            "ssl_version": getattr(ssl, self.tls_details.tls_version),
+                            "ciphers": self.tls_details.cipher})
+
+            try:
+                """
+                Establish connection with
+                a) Certificate based authorization
+                b) Certificate based authorization and username/password based authentication
+                """
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=self.host, port=self.port,
+                                                                                    credentials=credentials,
+                                                                                    ssl=True, ssl_options=ssl_details))
+                self.channel = self.connection.channel()
+            except Exception, e:
+                log.exception("Unable to establish connection")
+                raise e
+
+        else:
+            try:
+                """
+                Establish connection with
+                a) Plain AMQP
+                b) Username/password based authentication
+                """
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=self.host, port=self.port,
+                                                                                    credentials=credentials))
+                self.channel = self.connection.channel()
+            except Exception, e:
+                log.exception("Unable to establish connection")
+                raise e
+
+    def publish(self, exchange_name, routing_key, message, exchange_durable):
+        """
+        :param exchange_name: Exchange name
+        :type exchange_name: str or unicode
+        :param routing_key: Routing key for binding
+        :type routing_key: str or unicode
+        :param message: Message to be published
+        :type message: str or unicode
+        :param bool exchange_durable: Value to survive exchange on RabbitMQ reboot
+        :return:
+        """
+        # Declare exchange
+        self.channel.exchange_declare(exchange=exchange_name,
+                                      type='topic',
```

Comment:
I feel that we should not limit the capability of a protocol.  In IoT use case, `topic` will be most useful as it facilitates pub-sub.  If someone is chosing amqp instead of mqtt, there is a high chance that, they would like to leverage other mechanisms also.

We should keep it open and have `topic` as default.